adam-core/src/main/scala/org/bdgenomics/adam/rdd/feature/CoverageRDD.scala

```diff
@@ -91,8 +144,25 @@ case class CoverageRDD(rdd: RDD[Coverage],
 
   def union(rdds: CoverageRDD*): CoverageRDD = {
     val iterableRdds = rdds.toSeq
-    CoverageRDD(rdd.context.union(rdd, iterableRdds.map(_.rdd): _*),
-      iterableRdds.map(_.sequences).fold(sequences)(_ ++ _))
+
+    val mergedSequences = iterableRdds.map(_.sequences).fold(sequences)(_ ++ _)
+
+    if (iterableRdds.forall(rdd => rdd match {
+      case DatasetBoundCoverageRDD(_, _) => true
+      case _                             => false
+    })) {
+      DatasetBoundCoverageRDD(iterableRdds.map(_.dataset)
+        .fold(dataset)(_.union(_)), mergedSequences)
+    } else {
+      RDDBoundCoverageRDD(rdd.context.union(rdd, iterableRdds.map(_.rdd): _*),
```

Comment:
Ah, I see what you're saying. I think we should table this for later; I'll open an issue to track.